modules/tensor_mechanics/src/bcs/Pressure.C

```diff
@@ -30,15 +35,34 @@ Pressure::validParams()
 
 Pressure::Pressure(const InputParameters & parameters)
   : IntegratedBC(parameters),
-    _component(getParam<unsigned int>("component")),
+    _ndisp(coupledComponents("displacements")),
     _factor(getParam<Real>("factor")),
     _function(isParamValid("function") ? &getFunction("function") : NULL),
     _postprocessor(isParamValid("postprocessor") ? &getPostprocessorValue("postprocessor") : NULL),
     _alpha(getParam<Real>("alpha")),
-    _use_displaced_mesh(getParam<bool>("use_displaced_mesh"))
+    _fe_side(_assembly.getFEFace(_var.feType(), _sys.mesh().dimension())),
+    _q_dxi(nullptr),
+    _q_deta(nullptr),
+    _phi_dxi(nullptr),
+    _phi_deta(nullptr),
+    _use_displaced_mesh(getParam<bool>("use_displaced_mesh")),
+    _fe(libMesh::n_threads())
 {
-  if (_component > 2)
-    mooseError("Invalid component given for ", name(), ": ", _component, ".\n");
+  bool component_set = false;
+  for (unsigned int i = 0; i < _ndisp; ++i)
+  {
+    _disp_var.push_back(coupled("displacements", i));
+    if (_var.number() == _disp_var[i])
+    {
+      _component = i;
+      component_set = true;
+      if (parameters.isParamSetByUser("component") &&
+          _component != getParam<unsigned int>("component"))
+        mooseError("Incompatibility between component and displacements in " + _name);
+    }
+  }
+  if (!component_set)
+    mooseError("Problem with displacements in " + _name);
```

Comment:
in the initializer list:
```
    _component(libMesh::invalid_uint),
```

then
```suggestion
  for (unsigned int i = 0; i < _ndisp; ++i)
  {
    _disp_var.push_back(coupled("displacements", i));
    if (_var.number() == _disp_var[i])
    {
      _component = i;
      if (parameters.isParamSetByUser("component") &&
          _component != getParam<unsigned int>("component"))
        mooseError("Incompatibility between component and displacements in " + _name);
    }
  }
  if (_component == libMesh::invalid_uint)
    mooseError("Problem with displacements in " + _name);
```